### PR TITLE
Update aad webapp sample

### PR DESCRIPTION
**Pull request by yiliuTo**
Remove AAD backend samples and create webapp sample to support v2 endpoints only. The webapp sample project covers the following features of aad auto-configuration:
1. Access control on user groups.
2. Visit multiple resources.
3. On-demand authorization.

### CODE REVIEW

File: sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-webapp/src/main/java/com/azure/spring/sample/aad/security/AADOAuth2LoginSecurityConfig.java [hunk 1]
```diff
@@ -4,14 +4,22 @@
 package com.azure.spring.sample.aad.security;
 
 import com.azure.spring.aad.webapp.AzureOAuth2Configuration;
+import org.springframework.beans.factory.annotation.Autowired;
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-webapp/src/main/java/com/azure/spring/sample/aad/security/AADOAuth2LoginSecurityConfig.java:7]: Unused import, I think.

File: sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-webapp/src/main/resources/application.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      graph:
+        scopes: https://graph.microsoft.com/Calendars.Read
+      office:
+        scopes: https://manage.office.com/ActivityFeed.Read
+    client-id: <client-id>
+    client-secret: <client-secret>
+    tenant-id: <tenant-id>
+    user-group:
+      allowed-groups: group1, group2
+      value: '#microsoft.graph.group'
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-webapp/src/main/resources/application.yml:16]: `value` is not supported any more.

File: sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-webapp/src/main/java/com/azure/spring/sample/aad/controller/ClientController.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+public class ClientController {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ClientController.class);
+
+    @GetMapping("/graph")
+    @ResponseBody
+    public String graph(
+        @RegisteredOAuth2AuthorizedClient("graph") OAuth2AuthorizedClient oAuth2AuthorizedClient
+    ) {
+        logAuthorizedClient(oAuth2AuthorizedClient);
+        return "graph";
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-webapp/src/main/java/com/azure/spring/sample/aad/controller/ClientController.java:28]: We can return the json string of `oAuth2AuthorizedClient`, and output json string in `logAuthorizedClient`.

File: sdk/spring/azure-spring-boot-starter-active-directory/README.md [hunk 4]
```diff
@@ -290,8 +290,7 @@ For more information about setting logging in spring, please refer to the [offic
 The following section provides sample projects illustrating how to use the starter in different cases.
 ### More sample code
 - [Azure Active Directory for Frontend](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-resource-server)
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot-starter-active-directory/README.md:292]: `Azure Active Directory for WebApi`

File: sdk/spring/azure-spring-boot-starter-active-directory/README.md [hunk 5]
```diff
@@ -219,7 +219,7 @@ If you're using [Azure China](https://docs.microsoft.com/azure/china/china-welco
 azure.activedirectory.environment=cn-v2-graph
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot-starter-active-directory/README.md:219]: These will be deleted in another PR, right?

File: sdk/spring/azure-spring-boot-starter/README.md [hunk 6]
```diff
@@ -28,8 +28,7 @@ This starter brings auto configuration code for all Azure Spring modules, but to
 The following section provides sample projects illustrating how to use the Azure Spring Boot starters.
 ### More sample code
 - [Azure Active Directory for Frontend](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-resource-server)
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot-starter/README.md:30]: `Azure Active Directory for WebApi`

File: sdk/spring/azure-spring-boot-starter/README.md [hunk 7]
```diff
@@ -59,8 +58,7 @@ For more information about setting logging in spring, please refer to the [offic
...
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot-starter/README.md:60]: `Azure Active Directory for WebApi`

File: sdk/spring/azure-spring-boot/README.md [hunk 8]
```diff
@@ -59,8 +59,7 @@ For details, please see sample code in the [azure-spring-boot-sample-cloud-found
 The following section provides sample projects illustrating how to use the Azure Spring Boot starters.
 ### More sample code
 - [Azure Active Directory for Frontend](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-resource-server)
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot/README.md:61]: `Azure Active Directory for WebApi`

File: sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/webapp/AuthzCodeGrantRequestEntityConverter.java [hunk 9]
```diff
@@ -26,20 +26,17 @@ public AuthzCodeGrantRequestEntityConverter(AzureClientRegistration client) {
...
         RequestEntity<?> result = super.convert(request);
-        if (isRequestForDefaultClient(request)) {
-            Optional.ofNullable(result)
-                    .map(HttpEntity::getBody)
-                    .map(b -> (MultiValueMap<String, String>) b)
-                    .ifPresent(body -> body.add("scope", scopeValue()));
-        }
+        String scopes = String.join(" ", isRequestForDefaultClient(request) ?
+            azureClient.getAccessTokenScopes() :
+            request.getClientRegistration().getScopes());
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/webapp/AuthzCodeGrantRequestEntityConverter.java:31]: Put `?` and `:` at the beginning of a line.

**Review by chenrujun — CHANGES_REQUESTED** [review]
Please fix the pipeline failure.

File: sdk/spring/azure-spring-boot-starter-active-directory/README.md (hunk 5 above)
Comment (yiliuTo) [reply sdk/spring/azure-spring-boot-starter-active-directory/README.md:219]: There will be a PR later to update readme and changelog, but in case I forget this point, let me fix it now.

File: sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-webapp/src/main/java/com/azure/spring/sample/aad/controller/ClientController.java [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+@Controller
+public class ClientController {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ClientController.class);
+
+    @GetMapping("/graph")
+    @ResponseBody
+    public String graph(
+        @RegisteredOAuth2AuthorizedClient("graph") OAuth2AuthorizedClient oAuth2AuthorizedClient
+    ) throws JsonProcessingException {
+        ObjectMapper mapper = new ObjectMapper().enable(SerializationFeature.INDENT_OUTPUT);
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-webapp/src/main/java/com/azure/spring/sample/aad/controller/ClientController.java:26]: Can we set it as 
```
private static final ObjectMapper OBJECT_MAPPER= new ObjectMapper().enable(SerializationFeature.INDENT_OUTPUT);
```
And it's used in many classes, so write a `public static final toJsonString(Object) {...}` a util class is a better way.

File: sdk/spring/azure-spring-boot-starter-active-directory/README.md [hunk 11]
```diff
@@ -88,11 +88,11 @@ Refer to different samples for different authentication ways.
...
 **Note**: `AADAppRoleStatelessAuthenticationFilter` and `AADAuthenticationFilter` will be deprecated. [Click here](https://github.com/Azure/azure-sdk-for-java/issues/17860) to replace it.
 
-### Authenticate in backend
+### Authenticate in webapp
 
-Please refer to [azure-spring-boot-sample-active-directory-backend](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-backend/README.md) for authenticate in backend. Or [azure-spring-boot-sample-active-directory-backend-v2](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-backend-v2/README.md) to use Microsoft Graph API instead of Azure Active Directory Graph API.
+Please refer to [azure-spring-boot-sample-active-directory-webapp](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-webapp/README.md) for authenticate in backend.
 
-####  Configure application.properties:
+####  Configure application.properties:`
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot-starter-active-directory/README.md:95]: The diff is caused by mistake?

File: sdk/spring/azure-spring-boot-starter-active-directory/README.md [hunk 12]
```diff
@@ -214,12 +214,7 @@ spring.security.oauth2.client.provider.azure.jwk-set-uri=https://login.microsoft
 spring.security.oauth2.client.registration.azure.scope=openid, https://graph.windows.net/user.read, {your-customized-scope}
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot-starter-active-directory/README.md:214]: configuration items like `spring.security.oauth2.client.registration.azure.xxx` is not supported, it's OK to fix in another PR.

**Review by chenrujun — CHANGES_REQUESTED** [review]

File: sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-webapp/src/main/java/com/azure/spring/sample/aad/controller/ClientController.java (hunk 10 above)
Comment (yiliuTo) [reply sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-webapp/src/main/java/com/azure/spring/sample/aad/controller/ClientController.java:26]: Thanks, a good idea!

**Review by chenrujun — APPROVED** [review]
